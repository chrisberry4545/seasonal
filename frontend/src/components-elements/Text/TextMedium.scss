@import './../../styles/settings';
@import './../../styles/tools';

.c-text-medium {
  font-weight: 300;
  font-size: 0.9rem;
  color: $primary-text-color;
}
