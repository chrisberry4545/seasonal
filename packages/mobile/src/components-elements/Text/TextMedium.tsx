import React, { FC } from 'react';
import { Text, TextStyle } from 'react-native';
import { textBase } from './SharedTextStyles';

import {
  IText
} from './Text.interface';

export const styleTextMedium: TextStyle = {
  ...textBase,
  fontSize: 14,
  lineHeight: 14
};

export const TextMedium: FC<IText> = ({
  children,
  style
}) => (
  <Text style={ [styleTextMedium, style] }>
    { children }
  </Text>
);
