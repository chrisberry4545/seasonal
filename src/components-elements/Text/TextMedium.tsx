import React, { SFC } from 'react';
import { Text, TextStyle } from 'react-native';
import { textBase } from './SharedTextStyles';

import {
  ITextInterface
} from './Text.interface';

export const styleTextMedium: TextStyle = {
  ...textBase,
  fontSize: 14,
  lineHeight: 14
};

export const TextMedium: SFC<ITextInterface> = ({
  children,
  style
}) => (
  <Text style={ [styleTextMedium, style] }>
    { children }
  </Text>
);
