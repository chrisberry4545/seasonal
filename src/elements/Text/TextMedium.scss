.c-text-medium {
  font-weight: 300;
  font-size: 0.9rem;
}
