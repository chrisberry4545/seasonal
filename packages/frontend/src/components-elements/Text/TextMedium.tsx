import React, { FC } from 'react';
import './TextMedium.scss';
import { ITextInterface } from './Text.interface';

export const TextMedium: FC<ITextInterface> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-medium`}>
    { children }
  </span>
);
