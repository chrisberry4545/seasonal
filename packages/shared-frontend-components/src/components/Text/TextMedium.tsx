import React, { FC } from 'react';
import './TextMedium.scss';
import { IText } from './Text.interface';

export const TextMedium: FC<IText> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-medium`}>
    { children }
  </span>
);
