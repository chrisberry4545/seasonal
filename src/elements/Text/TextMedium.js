import React from 'react';
import './TextMedium.scss';

export const TextMedium = ({ className, children }) => (
  <span className={`${(className || '')} c-text-medium`}>
    { children }
  </span>
);
