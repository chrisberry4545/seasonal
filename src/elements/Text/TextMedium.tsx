import React, { SFC } from 'react';
import './TextMedium.scss';
import { ITextInterface } from './Text.interface';

export const TextMedium: SFC<ITextInterface> = ({
  className,
  children
}) => (
  <span className={`${(className || '')} c-text-medium`}>
    { children }
  </span>
);
