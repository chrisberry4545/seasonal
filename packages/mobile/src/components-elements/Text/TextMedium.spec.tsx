import React from 'react';
import { TextMedium } from './TextMedium';
import { shallow, ShallowWrapper } from 'enzyme';

jest.mock('react-native', () => ({
  Text: () => 'Text'
}));

describe('<TextMedium />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <TextMedium
        style={{ color: '#000000' }}>
          Text
      </TextMedium>
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());
});
