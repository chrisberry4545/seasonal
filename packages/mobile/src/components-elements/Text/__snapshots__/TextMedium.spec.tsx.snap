// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextMedium /> renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#20303C",
        "fontFamily": "Roboto",
        "fontSize": 14,
        "lineHeight": 14,
      },
      Object {
        "color": "#000000",
      },
    ]
  }
>
  Text
</Text>
`;
